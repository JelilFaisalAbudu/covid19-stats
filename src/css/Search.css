.search-wrapper {
  display: flex;
  align-items: stretch;
  margin: 1em;
}

.search-wrapper > input {
  flex-grow: 1;
}
