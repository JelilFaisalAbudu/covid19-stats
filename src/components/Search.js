import PropTypes from 'prop-types';
import '../css/Search.css';

const Search = ({ query, handleChange, children }) => (
  <div className="search-wrapper">
    <input
      type="text"
      className="form-control"
      id="inputSearch"
      value={query}
      onChange={handleChange}
      placeholder={children}
    />
  </div>
);

Search.propTypes = {
  query: PropTypes.string,
  handleChange: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
};

Search.defaultProps = { query: '' };

export default Search;
