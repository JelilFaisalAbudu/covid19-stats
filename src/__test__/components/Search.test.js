import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { shallow } from 'enzyme';
import '../setup/setupTests';
import Search from '../../components/Search';

const handleChange = query => query;

const setup = () => shallow(<Search handleChange={handleChange} />);

describe('Search component', () => {
  let component;
  beforeEach(() => {
    component = setup();
  });

  it('Should render', () => {
    const element = component.find('.form-control');
    expect(element.length).toEqual(1);
  });

  test('renders the search bar in the page', () => {
    const item = component.find('.search-wrapper');
    expect(item).toMatchSnapshot();
  });
});
