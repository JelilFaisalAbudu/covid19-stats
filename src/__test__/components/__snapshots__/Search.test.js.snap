// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders the search bar in the page 1`] = `ShallowWrapper {}`;
